t/t-clone.sh

```diff
@@ -178,6 +178,33 @@ begin_test "clone ClientCert"
     git config --global "http.sslBackend" "openssl"
   fi
 
+  # Note that the record files we create in the $CREDSDIR directory are not
+  # used until we set the "http.sslCertPasswordProtected" option to "true"
+  # and the "http.<url>.sslKey" option with the path to our TLS/SSL client
+  # certificate's encrypted private key file.  (The PEM certificate file
+  # itself is not encrypted and does not contain the private key.)
+  #
+  # When these options are set, however, Git and Git LFS will independently
+  # invoke "git credential fill" to retrieve the passphrase for the
+  # encrypted private key.  Because the "http.sslCertPasswordProtected"
+  # option is set, Git will query the credential helper, passing a
+  # "protocol=cert" line and a "path=<certfile>" line with the path
+  # from the "http.<url>.sslCert" option.  Note that this path refers
+  # to our unencrypted certificate file; Git does not use the path to
+  # the encrypted private key file from the "http.<url>.sslKey" option
+  # in its query to the credential helper.
```

Comment:
Is that a bug in Git? Why does Git query for the cert file and not the key file?

Reply:
Well, it's behaviour that stems from this [line](https://github.com/git/git/blob/090d24e9af6e9f59c3f7bee97c42bb1ae3c7f559/http.c#L698) where the certificate file's path, not the key file's path, is sent to the credential helper, and it's clearly [documented](https://github.com/git/git/blob/090d24e9af6e9f59c3f7bee97c42bb1ae3c7f559/Documentation/config/http.txt#L198-L202), because the certificate file *may* be encrypted, since those PEM files often contain both the certificate and the private key.  From git/git@30dd916348001e4313708473d91d633d3b14d1b5 there's this explanation, as well:

> The user is always prompted, even if the certificate is not encrypted.  This should be fine; unencrypted certificates are rare and a security risk anyway.

In our `lfstest-gitserver` test utility, we only [encrypt](https://github.com/git-lfs/git-lfs/blob/3990c7ab4fa6c99540af3f73f33da47a72df08af/t/cmd/lfstest-gitserver.go#L1768) the private key, and the client certificate file we create does not include it, so there's no particular need to encrypt that file.

To your point, the Git LFS client should arguably behave more like Git, and pass the file to the certificate file, not the key file.  At the moment, we simply expect to be able to [read](https://github.com/git-lfs/git-lfs/blob/3990c7ab4fa6c99540af3f73f33da47a72df08af/lfshttp/certs.go#L87) the client certificate file, but [check](https://github.com/git-lfs/git-lfs/blob/3990c7ab4fa6c99540af3f73f33da47a72df08af/lfshttp/certs.go#L102) if the private key file was encrypted, and if it was, [call](https://github.com/git-lfs/git-lfs/blob/3990c7ab4fa6c99540af3f73f33da47a72df08af/lfshttp/certs.go#L103) a function and pass the key file's path to it, so it [uses](https://github.com/git-lfs/git-lfs/blob/3990c7ab4fa6c99540af3f73f33da47a72df08af/lfshttp/certs.go#L43) that path in the call to the credential helper.  Changing that is outside the scope of this PR, but it's worth noting in case we want to adjust it in the future.

I think having the extra comments in these tests is a good step in that direction, since it makes it clear what's going on and the differences between Git and Git LFS in this regard.